pkg/credentialprovider/config.go

```diff
@@ -64,6 +74,32 @@ func GetPreferredDockercfgPath() string {
 }
 
 func ReadDockerConfigFile() (cfg DockerConfig, err error) {
+	// Try happy path first - latest config file
```

Comment:
How is docker handling the transition?  Are the files complementary?  If so, we should probably have two different provider instances to build the keyring.